frontend/components/ManageStakingPage/StakingContractSection/MigrateButton.tsx

```diff
@@ -77,25 +87,27 @@ export const MigrateButton = ({ stakingProgramId }: MigrateButtonProps) => {
           setIsBalancePollingPaused(true);
 
           try {
-            setServiceStatus(MiddlewareDeploymentStatus.DEPLOYING);
+            setDeploymentStatus(MiddlewareDeploymentStatus.DEPLOYING);
             goto(Pages.Main);
 
-            await ServicesService.createService({
-              stakingProgramId,
+            // TODO: create type for this response, we need the service_config_id to update the relevant service
+            // eslint-disable-next-line @typescript-eslint/no-unused-vars
+            const createServiceResponse = await ServicesService.createService({
+              stakingProgramId: stakingProgramIdToMigrateTo,
               serviceTemplate,
               deploy: true,
               useMechMarketplace: false,
             });
 
-            await updateStakingProgram();
+            await updateStakingProgram(); // TODO: refactor to support single staking program & multi staking programs, this on longer works
```

Comment:
`updateStakingProgram` immediately refetches the staking program blockchain state, instead of wait for the poll

as we'll be supporting multiple agents & chains, we can pass stakingProgramId (from props), to update the specific program